File: evd_ros_backend/evd_ros_core/src/evd_script/program/tasks.py/simple_pick_and_place.py
from ..task import Task
from ..primitives import MoveTrajectory
from .open_gripper import OpenGripper
from .close_gripper import CloseGripper


class SimplePickAndPlace(Task):

    '''
    Data structure methods
    '''

    def __init__(self, startLocUuid=None, pickLocUuid=None, placeLocUuid=None, type='', name='',
                 uuid=None, parent=None, append_type=True, create_default=True, primitives=None, context=None):

        if primitives == None:
            primitives = [
                MoveTrajectory(startLocUuid,pickLocUuid,create_default=create_default),
                CloseGripper(),
                MoveTrajectory(pickLocUuid,placeLocUuid,create_default=create_default),
                OpenGripper()
            ]

        super(PickAndPlace,self).__init__(
            type='simple-pick-and-place.'+type if append_type else type,
            name=name,
            uuid=uuid,
            parent=parent,
            append_type=append_type,
            context=context,
            primitives=primitives)

File: evd_authoring_app/src/model/evdScript/program/tasks/simplePickAndPlace.js
import { MoveTrajectory } from '../primitives';
import { CloseGripper } from './closeGripper';
import { OpenGripper } from './openGripper';
import { Task } from '../task';
import { NodeParser } from '../../utilityFunctions';


export class SimplePickAndPlace extends Task {

    /*
    * Data structure methods
    */

    static typeString() {
        return 'simple-pick-and-place';
    }

    static fullTypeString() {
        return Task.fullTypeString() + SimplePickAndPlace.typeString();
    }

    constructor(startLocUuid=null, pickLocUuid=null, placeLocUuid=null, thingUuid=null, 
                type='', name='', uuid=null, parent=null, appendType=true, primitives=null) 
    {

        if (primitives === null) {
            primitives = [
                MoveTrajectory(startLocUuid,pickLocUuid),
                CloseGripper(thingUuid),
                MoveTrajectory(pickLocUuid,placeLocUuid),
                OpenGripper(thingUuid)
            ];
        }

        super(
            primitives,
            (appendType) ? 'simple-pick-and-place.'+type : type,
            name,
            uuid,
            parent,
            appendType
        );
    }

    static fromDict(dct) {
        return new SimplePickAndPlace(
            null,
            null,
            null,
            null,
            dct.type,
            dct.name,
            dct.uuid,
            null,
            false,
            dct.primitives.map(p => NodeParser(p))
        );
    }
}